graphql_compiler/query_formatting/gremlin_formatting.py

```diff
@@ -48,6 +49,22 @@ def _safe_gremlin_string(value):
     return final_escaped_value
 
 
+def _safe_gremlin_decimal(value):
+    """Represent decimal objects as Gremlin strings."""
+    if isinstance(value, Decimal):
+        # We got a decimal object, serialize it directly.
+        # The "G" suffix on a decimal number forces it to be a BigInteger/BigDecimal literal:
+        # http://docs.groovy-lang.org/next/html/documentation/core-syntax.html#_number_type_suffixes
+        return str(value) + 'G'
+    else:
+        # We didn't get a decimal object, try to make it into one and then serialize it.
```

Comment:
In what situation does this happen?

Reply:
If e.g. a user used a JSON-based endpoint to submit a GraphQL query, since JSON doesn't support `Decimal` natively, they may have used a `float` or `string` representation instead. If the `Decimal` constructor is able to parse the number correctly, we do that just to be easier on the user since we already have type metadata available.